mix/ContractCallDataEncoder.cpp

```diff
@@ -55,54 +63,91 @@ void ContractCallDataEncoder::encode(QFunctionDefinition const* _function)
 }
 
+void ContractCallDataEncoder::encodeArray(QJsonArray const& _array, QList<int> _dim, SolidityType const& _type, bytes& _content)
+{
+	size_t offsetStart = _content.size();
+	if (_dim[0] == -1)
+	{
+		bytes count = bytes(32);
+		toBigEndian((u256)_array.size(), count);
+		_content += count; //reserved space for count
+	}
+	_dim.pop_front();
+
+	int k = 0;
+	for (QJsonValue const& c: _array)
+	{
+		if (c.isArray())
+		{
+			if (_dim[0] == -1)
+				m_dynamicOffsetMap.push_back(std::make_pair(m_dynamicData.size() + offsetStart + 32 + k * 32, m_dynamicData.size() + _content.size()));
+			encodeArray(c.toArray(), _dim, _type, _content);
+		}
+		else
+		{
+			// encode single item
+			if (c.isDouble())
+				encodeSingleItem(QString::number(c.toDouble()), _type, _content);
+			else if (c.isString())
+				encodeSingleItem(c.toString(), _type, _content);
+		}
+		k++;
+	}
+}
+
 void ContractCallDataEncoder::encode(QVariant const& _data, SolidityType const& _type)
 {
-	u256 count = 1;
 	QStringList strList;
 	if (_type.array)
 	{
-		if (_data.type() == QVariant::String)
-			strList = _data.toString().split(",", QString::SkipEmptyParts);  //TODO: proper parsing
-		else
-			strList = _data.toStringList();
-		count = strList.count();
+		QList<int> dim = extractDimension(_type. name);
+		bytes content;
+		QJsonDocument jsonResponse = QJsonDocument::fromJson(_data.toString().toUtf8());
+		QJsonArray jsonObject = jsonResponse.array();
+		size_t size = m_encodedData.size();
+		if (dim[0] == -1)
+		{
+			m_encodedData += bytes(32); // reserve space for offset
+			m_staticOffsetMap.push_back(std::make_pair(size, m_dynamicData.size()));
+		}
+		encodeArray(jsonObject, dim, _type, content);
 
+		if (!_type.dynamicSize)
+			m_encodedData.insert(m_encodedData.end(), content.begin(), content.end());
+		else
+			m_dynamicData.insert(m_dynamicData.end(), content.begin(), content.end());
 	}
-	else
-		strList.append(_data.toString());
-
-	if (_type.dynamicSize)
+	else if (_type.dynamicSize && _type.type == SolidityType::Type::Bytes)
 	{
 		bytes empty(32);
 		size_t sizePos = m_dynamicData.size();
 		m_dynamicData += empty; //reserve space for count
-		if (_type.type == SolidityType::Type::Bytes)
-			count = encodeSingleItem(_data.toString(), _type, m_dynamicData);
-		else
-		{
-			count = strList.count();
-			for (auto const& item: strList)
-				encodeSingleItem(item, _type, m_dynamicData);
-		}
+		u256 count = encodeSingleItem(_data.toString(), _type, m_dynamicData);
 		vector_ref<byte> sizeRef(m_dynamicData.data() + sizePos, 32);
 		toBigEndian(count, sizeRef);
-		m_offsetMap.push_back(std::make_pair(m_encodedData.size(), sizePos));
+		m_staticOffsetMap.push_back(std::make_pair(m_encodedData.size(), sizePos));
 		m_encodedData += empty; //reserve space for offset
 	}
 	else
+		encodeSingleItem(_data.toString(), _type, m_encodedData);
+}
+
+QList<int> ContractCallDataEncoder::extractDimension(QString const& _type)
```

Comment:
it was easier for now to parse string. will change it when the dimension will be available in SolidityType or in the ABI.